File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int solution(int[][] triangle) {
        List<Integer> list = new ArrayList<Integer>();
        int answer=0;

        int fn_0=0;// f(0)
        int fn_1=1;// f(1)
        int count =0;
        int sum =0;
        for(int i = triangle.length-2; i >= 0; i--) {
            for(int j = 0 ; j< triangle[i].length; j++) {

                    fn_0 = triangle[i+1][j];
                    fn_1 = triangle[i+1][j+1];
                    triangle[i][j] = Math.max(fn_0 + triangle[i][j],fn_1 + triangle[i][j]);
                    //System.out.println(triangle[i][j]);
                    answer = triangle[i][j];
            }
        }

        return answer;

    }
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp

#include <iostream>
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> triangle) {
    int answer = 0;
    
    vector<vector<int>> dp(triangle.size(), vector<int>(triangle.size()));
    
    dp[0][0] = triangle[0][0];
    
    for(int i = 1; i < triangle.size(); i++)
    {
        for(int j = 0; j < triangle[i].size(); j++) 
        {
            dp[i][j] = max(triangle[i][j] + dp[i - 1][j], triangle[i][j] + dp[i - 1][j - 1]);
        }
    }
    
    for(int i = 0; i < triangle[triangle.size() - 1].size(); i++)
    {
        answer = max(answer , dp[triangle.size() - 1][i]);
    }
    
    return answer;
}